src/NHibernate/Engine/TableGroupJoinHelper.cs

```diff
@@ -68,14 +68,15 @@ private static bool NeedsTableGroupJoin(IReadOnlyList<IJoin> joins, SqlString[]
 
 			foreach (var join in joins)
 			{
-				var entityPersister = GetEntityPersister(join.Joinable);
+				var entityPersister = GetEntityPersister(join.Joinable, out var collection);
```

Comment:
Maybe it is better to return `isManyToMany` as a second parameter here?

Reply:
Ok